### ISSUE TITLE
Export gradient boosting models trained in Elastic stack

### ISSUE THREAD

**Issue by valeriy42** [elastic/eland#503]
In Elastic Stack we can train supervised learning models to create an ensemble of gradient-boosting trees. This issue suggests extending the codebase of eland to be able to download these models on a local machine as an instance of [`sklearn.Pipeline`](https://scikit-learn.org/stable/modules/compose.html):
```python
es_model = eland.ml.MLModel(es_client=es_client, model_id=model_id)
pipeline = es_model.export_model()
pipeline.fit(X)
```

## Why?

Being able to treat the ES models as `sklearn` pipelines will allow data scientists to use their usual tools to analyze, explore, or debug the models, prove unbiasedness, etc. 

## How?

We can convert the model JSON definition into a subclass of  `GradientBoostingClassifier` or `GradientBoostingRegressor` of `sklearn`. The encoding preprocessor can be represented as `ColumnTransformer`.  Once we have these individual components, we can use `sklearn.Pipeline` to produce a single object that can be evaluated by passing the original data frame.

We will be able to run the functionality similarly and obtain numerically similar results:
```python
es_prediction = es_model.predict_proba(X, feature_names_in=pipeline["preprocessor"].get_feature_names_out())
sklearn_prediction = pipeline.predic_proba(X, feature_names_in=pipeline["preprocessor"].get_feature_names_out())

numpy.tsting.assert_allclose(es_prediction, sklearn_prediction)
```

## Limitations

Since we have to instantiate the model locally, we will be limited by the memory size and computational capacity of a laptop/desktop. We will document this limitation, although most models should be able to fit into 8-16GB of memory available on modern PCs.
